classes/class-admin-notices.php

```diff
@@ -0,0 +1,1007 @@
+ */
+final class Admin_Notices extends Base {
+
+    /**
+	 * Register the loader for the admin notices.
+	 */
+    public function __construct() {
+		parent::__construct();
+
+        \add_action( 'load-index.php', array( $this, 'load_notices' ) );
+        \add_action( 'load-upload.php', array( $this, 'load_notices' ), 9 );
+		\add_action( 'load-media-new.php', array( $this, 'load_notices' ) );
+		\add_action( 'load-media_page_ewww-image-optimizer-bulk', array( $this, 'load_notices' ) );
+		\add_action( 'load-plugins.php', array( $this, 'load_notices' ) );
+		
+		// Prevent Autoptimize from displaying its image optimization notice.
+		\remove_action( 'admin_notices', 'autoptimizeMain::notice_plug_imgopt' );
+    }
+
+    /**
+	 * Register the admin notices, and add the actions to display them.
+	 */
+    public function load_notices() {
+        \add_action( 'network_admin_notices', array( $this, 'display_notices' ) );
+        \add_action( 'admin_notices', array( $this, 'display_notices' ) );
+    }
+
+	/**
+	 * Register the handlers for dismissing notices.
+	 */
+	public function register_action_handlers() {
+		// AJAX action hook to dismiss the UTF-8 notice.
+		\add_action( 'wp_ajax_ewww_dismiss_utf8_notice', array( $this, 'dismiss_utf8_notice' ) );
+		// AJAX action hook to dismiss the exec notice and other related notices.
+		\add_action( 'wp_ajax_ewww_dismiss_exec_notice', array( $this, 'dismiss_exec_notice' ) );
+		// AJAX action hook to dismiss the WooCommerce regen notice.
+		\add_action( 'wp_ajax_ewww_dismiss_wc_regen', array( $this, 'dismiss_wc_regen_notice' ) );
+		// AJAX action hook to dismiss the WP/LR Sync regen notice.
+		\add_action( 'wp_ajax_ewww_dismiss_lr_sync', array( $this, 'dismiss_lightroom_sync_notice' ) );
+		// AJAX action hook to disable the media library notice.
+		\add_action( 'wp_ajax_ewww_dismiss_media_notice', array( $this, 'dismiss_media_notice' ) );
+		// AJAX action hook to disable the 'review request' notice.
+		\add_action( 'wp_ajax_ewww_dismiss_review_notice', array( $this, 'dismiss_review_notice' ) );
+		// AJAX action hook to disable the newsletter signup banner.
+		\add_action( 'wp_ajax_ewww_dismiss_newsletter', array( $this, 'dismiss_newsletter_signup_notice' ) );
+	}
+    
+    /**
+     * Hook to display notices on the admin pages.
+     */
+    public function display_notices() {
+        $admin_permissions = apply_filters( 'ewww_image_optimizer_admin_permissions', '' );
+		if ( current_user_can( $admin_permissions ) ) {
+			$this->test_mode_notice();
+			$this->debug_mode_notice();
+            $this->utils_notice();
+            $this->os_notice();
+            $this->hosting_requires_api_notice();
+			$this->utf8_db_notice();
+			$this->upgrade_620_notice();
+			$this->invalid_key_notice();
+			$this->schedule_noasync_notice();
+			$this->php_notice();
+			$this->exactdn_hmwp_notice();
+			$this->review_appreciated_notice();
+			$this->thumbnail_regen_notice();
```

Comment:
The thumb regen notices won't trigger via this route, because they'll never run on the actual thumb regen pages.
This one need an add_action() call in the __construct() method, like so:
`\add_action( 'admin_notices', array( $this, 'thumbnail_regen_notice' ) );`